app/_kong_plugins/openid-connect/index.md

```diff
@@ -37,6 +35,541 @@ search_aliases:
+is likely when the OpenID Connect plugin is configured to use one client, and the refresh token is retrieved
+with another. 
+
+The grant itself is very similar to the [password grant](#password-grant-workflow) and
+the [client credentials grant](#client-credentials-grant-workflow):
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant idp as IdP <br>(e.g. Keycloak)
+    participant httpbin as Upstream <br>(upstream service,<br> e.g. httpbin)
+    activate client
+    activate kong
+    client->>kong: service with<br>refresh token
+    deactivate client
+    kong->>kong: load refresh token
+    activate idp
+    kong->>idp: IdP/token with<br>client credentials and<br>refresh token
+    deactivate kong
+    idp->>idp: authenticate client and<br>verify refresh token
+    activate kong
+    idp->>kong: return tokens
+    deactivate idp
+    kong->>kong: verify tokens
+    activate httpbin
+    kong->>httpbin: request with access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+### Session authentication workflow
+
+The OpenID Connect plugin can issue a session cookie that can be used for further
+session authentication. To make OpenID Connect issue a session cookie, you need
+to first authenticate with one of the other grants or flows that this plugin supports. 
+For example, the [authorization code flow](#authorization-code-flow)
+demonstrates session authentication when it uses the redirect login action.s
+
+The session authentication portion of the flow works like this:
+
+<!--vale off-->
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant httpbin as Upstream <br>(upstream service,<br> e.g. httpbin)
+    activate client
+    activate kong
+    client->>kong: service with<br>session cookie
+    deactivate client
+    kong->>kong: load session cookie
+    kong->>kong: verify session
+    activate httpbin
+    kong->>httpbin: request with<br>access token
+    httpbin->>kong: response
+    deactivate httpbin
+    activate client
+    kong->>client: response
+    deactivate kong
+    deactivate client
+{% endmermaid %}
+<!--vale on-->
+
+### User info authentication flow
+
+The user info authentication uses OpenID Connect standard user info endpoint to verify the access token.
+In most cases, you would use [introspection authentication](#introspection-authentication-flow) instead of user info,
+as introspection is meant for retrieving information from the token itself, whereas the user info endpoint is
+meant for retrieving information about the user to whom the token was given. The flow is almost identical to introspection authentication:
```

Comment:
```suggestion
The user info authentication uses OpenID Connect standard user info endpoint to verify the access token.
In most cases, you would use [introspection authentication](#introspection-authentication-flow) instead of user info, as introspection is meant for retrieving information from the token itself, whereas the user info endpoint is meant for retrieving information about the user to whom the token was given. The flow is almost identical to introspection authentication:
```